doc/adr/0007-deprecate-servicesync.md

```diff
@@ -0,0 +1,48 @@
+# 7. Deprecate ServiceSync
+
+Date: 2024-01-16
+
+## Status
+
+Proposed
+
+Enabled By [6. Explicit resources for declaring service bindings](0006-explicit-resources-for-declaring-service-bindings.md)
+
+## Context
+
+ServiceSync is a mechanism whereby service ingress bindings are
```

Comment:
Service synchronisation (ServiceSync) is a feature that creates listeners  on every site 
when a service is exposed. This feature does not require user actions on those sites.

In other words, ServiceSync is a mechanism whereby service ingress bindings are